service/src/test/java/bio/terra/lz/futureservice/pact/provider/LandingZoneServicePactProviderTest.java

```diff
@@ -0,0 +1,260 @@
+import bio.terra.lz.futureservice.generated.model.ApiAzureLandingZoneResourcesList;
+import bio.terra.lz.futureservice.generated.model.ApiAzureLandingZoneResourcesPurposeGroup;
+import bio.terra.lz.futureservice.generated.model.ApiAzureLandingZoneResult;
+import bio.terra.lz.futureservice.generated.model.ApiCreateLandingZoneResult;
+import bio.terra.lz.futureservice.generated.model.ApiDeleteAzureLandingZoneJobResult;
+import bio.terra.lz.futureservice.generated.model.ApiDeleteAzureLandingZoneResult;
+import bio.terra.lz.futureservice.generated.model.ApiErrorReport;
+import bio.terra.lz.futureservice.generated.model.ApiJobReport;
+import bio.terra.lz.futureservice.generated.model.ApiResourceQuota;
+import java.time.Instant;
+import java.time.OffsetDateTime;
+import java.time.ZoneOffset;
+import java.util.List;
+import java.util.Map;
+import java.util.UUID;
+import org.apache.commons.lang3.StringUtils;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Tag;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.AutoConfigureMockMvc;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.test.web.servlet.MockMvc;
+
+/**
+ * To run these tests against a local set of pacts, comment out the @PactBroker annotation and
+ * replace with an @PactFolder annotation with a path to your pacts
+ */
+@AutoConfigureMockMvc
+@Tag("pact-provider-test")
+@Provider("terra-landing-zone-service")
+@PactBroker
+// @PactFolder("/path/to/pacts")
+@SpringBootTest(
+    properties = {"otel.instrumentation.spring-webmvc.enabled=false"},
+    classes = LandingZoneApplication.class)
+public class LandingZoneServicePactProviderTest extends BaseSpringUnitTest {
+  private static final String CONSUMER_BRANCH = System.getenv("CONSUMER_BRANCH");
+
+  @MockBean BearerTokenFactory bearerTokenFactory;
+  @MockBean LandingZoneAppService landingZoneAppService;
+  @Autowired private MockMvc mockMvc;
+
+  @PactBrokerConsumerVersionSelectors
+  public static SelectorBuilder consumerVersionSelectors() {
+    // The following match condition basically says
+    // If verification is triggered by Pact Broker webhook due to consumer pact change, verify only
+    // the changed pact.
+    // Otherwise, this is a PR, verify all consumer pacts in Pact Broker marked with a deployment
+    // tag (e.g. dev, alpha).
+    if (StringUtils.isBlank(CONSUMER_BRANCH)) {
+      return new SelectorBuilder().mainBranch().deployedOrReleased();
+    } else {
+      return new SelectorBuilder().branch(CONSUMER_BRANCH);
+    }
+  }
+
+  @BeforeEach
+  void setUp(PactVerificationContext context) {
+    context.setTarget(new MockMvcTestTarget(mockMvc));
+    when(landingZoneAppService.deleteLandingZone(any(), any(), any(), any()))
```

Comment:
I'm a bit squirrelly about this much mock setup being done for every test case. 

The most articulatable reason I have, is that it doesn't provide a way for consumers to use provider side parameters, so they'll probably end up using the values here. This works, but it makes the tests more brittle, and the resulting contracts more specific.

Reply:
Yeah good point. Most of these fell under the "given an existing landing zone" state so I've moved the mocking to that state method.